content/en/blog/ksynth-k8s-operator.md

```diff
@@ -0,0 +1,134 @@
+on them as well?
+
+## Odyssey
+[Odyssey](https://github.com/kentik/odyssey) is a simple Kubernetes operator that automates
+agent provisioning and configuration for a synthetic test framework in a Kubernetes cluster. The
+[operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) enables
+developers to extend Kubernetes for their own applications and still follow the Kubernetes principles.
+When used with Kentik Synthetics, this gives a number of benefits:
+
+- Standard synthetic platform for all teams
+- Operator controlled base framework
+- Cloud provider access managed by operator not team
+- Developers focus on writing checks not managing infra
+
+Operators deploy and manage the Odyssey operator enabling teams to focus on writing tests. It also
+improves security by keeping provider credentials at the controller level and not alongside the application.
+By leveraging the teams existing Kubernetes knowledge they have a much more familiar and simpler
+method to add synthetic checks to their applications.
+
+## Checks
+Once Odyssey is deployed, adding checks to an application or service is simple and familiar. Odyssey provides
+a [custom resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to
+enable simple check creation. Let's look at how to add a simple HTTP check to an application:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: demo
+  labels:
+    app: demo
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: demo
+  template:
+    metadata:
+      labels:
+        app: demo
+    spec:
+      containers:
+      - name: demo
+        image: docker.io/ehazlett/sysinfo:latest
+        ports:
+          - containerPort: 8080
+            name: app
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: demo
+  labels:
+    app: demo
+spec:
+  selector:
+    app: demo
+  ports:
+    - name: app
+      port: 8080
+      targetPort: app
+      protocol: TCP
+```
+
+The above is a simple web service that provides node information. It contains a deployment with three replicas that each
+run a pod that provides a web service on port 8080. There is also a Kubernetes service that provides access
+to the pods also on port 8080. Now let's look at adding an Odyssey check to the application that performs a simple
+`GET`:
+
+```yaml
+apiVersion: synthetics.kentiklabs.com/v1
+kind: SyntheticTask
+metadata:
+  name: demo
+spec:
+  fetch:
+    - service: demo
+      target: /
+      port: 8080
+      method: GET
+      period: 10s
+      expiry: 5s
+```
+
+That's it. When the controller detects this resource is created, it will configure the agent to perform a fetch to the
+service `demo` on port 8080 every 10 seconds. Odyssey automatically resolves the `demo` service endpoint making the
+check work on any environment.
+
+Once we have defined our checks we can view them in the [Kentik Portal](https://www.kentik.com). This gives teams
```

Comment:
Worth it to go into more detail on how to define these checks in the portal? That's the part I would worry gets hard.

Reply:
These checks are all automated from the Odyssey controller.  There are no user defined checks in the portal for Odyssey - just via the spec.

Reply:
I will mention the other types though +1